test/morphology/test_black_hat.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+import torch
+import kornia.morphology as m
+import kornia.testing as utils  # test utils
+from torch.autograd import gradcheck
+from torch.testing import assert_allclose
+
+
+class TestBlackHat(utils.BaseTester):
+
+    def test_smoke(self, device, dtype):
+        kernel = torch.rand(3, 3, device=device, dtype=dtype)
+        assert m.se_to_mask(kernel) is not None
+
+    def test_batch(self, device, dtype):
+        input = torch.rand(3, 2, 6, 10, device=device, dtype=dtype)
+        kernel = torch.rand(3, 3, device=device, dtype=dtype)
```

Comment:
I suggest to add parametrize with different image and kernels size (check for even shapes as common corner cases). Check kornia color test_cardinality